src/Zio/UPath.cs

```diff
@@ -296,165 +277,159 @@ internal static StringBuilder GetSharedStringBuilder()
         var parts = internalHelper.Slices;
         parts.Clear();
 
-        var builder = internalHelper.Builder;
-        builder.Length = 0;
-
         var lastIndex = 0;
-        try
```

Comment:
removed the exception handling (as mentioned in the resetting of the builder comment), which caused the removal of the whitespaces, did not change the code itself except where called out.